### ISSUE TITLE
6.3-beta: navigation block CSS is not output on the front-end

### ISSUE THREAD

**Issue by pbiron** [WordPress/gutenberg#52391]
### Description

In WP 6.3-beta2 and 6.3-beta3, when navigation blocks are added to a template part, they render fine in the site editor, but not on the front end.  As far as I can tell, the CSS for them is **not** currently being output on the front end.

### Step-by-step reproduction instructions

1. set any block-based theme as active (I used TT3 and a bespoke block-based theme in my testing)
2. add a Navigation block (with a Page List child) to the header template part.
3. save the template part.
4. view the site on the front end and notice that the navigation block doesn't look right

### Screenshots, screen recording, code snippet

This is what the front end looks like
![wp-6 3-beta-navigation-block-front-end](https://github.com/WordPress/gutenberg/assets/3824560/bdf6b8c0-e05e-487f-9fb7-e025e927cb50)

This is what the site editor looks like
![wp-6 3-beta-navigation-block-site-editor](https://github.com/WordPress/gutenberg/assets/3824560/b2b31764-ccff-429d-8822-d63e37001ea5)

### Environment info

WP 6.3-beta2 and 6.3-beta3 (w/o GB plugin)
TT3 and a bespoke block-based theme
Windows 11, Chrome w/ a modified WAMP stack

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by jordesign** [comment]
Hi @pbiron  - I haven't yet been able to recreate this in my testing (using beta3 and TwentyTwentyTwo).

Just to confirm the structure of your menu in the test - is that a Page List inside the Navigation block? Just with two nested pages?

**Comment by pbiron** [comment]
> Just to confirm the structure of your menu in the test - is that a Page List inside the Navigation block? Just with two nested pages?

Correct.  The number of pages in the page list doesn't really matter, tho...

**Comment by pbiron** [comment]
I'm getting this behavior on every site I've upgraded from 6.2.2 to 6.3-beta{2,3}.  I also just created a new site with 6.3-beta3 and am also seeing this behavior on that site (so whatever the problem is, isn't dependent on any menu pre-existing in 6.2 before the update to 6.3).

I also just tested with TT2 and seeing it there as well (so the problem doesn't seem to depend on the particular block theme being used...it happens for with with any block theme).

**Comment by t-hamano** [comment]
Hi @pbiron,

This issue was also reported in https://github.com/WordPress/gutenberg/issues/52199, but it is probably specific to the Windows OS environment.

Here is the ticket for the core: https://core.trac.wordpress.org/ticket/58711

This problem has also been fixed in [Changeset 56136](https://core.trac.wordpress.org/changeset/56136). Therefore, the fix will be reflected in the nightly build or the next beta version.

I would like to close this issue, but if I have missed something, please comment.

**Comment by pbiron** [comment]
@t-hamano I just updated one site to 6-3 nightly (6.3-beta3-56155) and the problem continued.  I then looked at the changeset you mention and realized that the change is only operant if the `wp_core_block_css_files` transient` doesn't exist.  So, I deleted that transient, reloaded a page on the front end and **success**, the navigation menu renders correctly.

So, I'm closing this.  Thanx for noticing the similarity between this issue and the other one!
